test/stockpile_test.clj

```diff
@@ -160,6 +163,39 @@
            (finally
              (Files/delete dest))))))))
 
+(defn fill-filesystem [path]
+  (let [append StandardOpenOption/APPEND
+        buf (byte-array (* 64 1024) (byte \?))
+        write-chunks (fn [write-chunk open-opts]
+                       (with-open [out (Files/newOutputStream
+                                        path
+                                        (into-array OpenOption open-opts))]
+                         (try
+                           (while true (write-chunk out))
+                           (catch IOException ex true))))]
+    ;; Write smaller and smaller chunks; finish up with single bytes.
+    (write-chunks #(.write % buf 0 (* 64 1024)) [])
+    (write-chunks #(.write % buf 0 1024) [append])
+    (write-chunks #(.write % (int \?)) [append])
+    (let [free (.getUsableSpace (Files/getFileStore path))]
+      (is (= 0 free))
+      (zero? free))))
```

Comment:
Though they do document that for `thrown?`, but not `thrown-with-msg?`, so of course it could just be incomplete documentation: https://clojure.github.io/clojure/clojure.test-api.html#clojure.test/is